// office-ui-fabric-react/lib-amd/demo/pages/LayerPage/examples/Layer.Basic.Example.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'react', '../../../../index', './Layer.Example.scss'], function (require, exports, React, index_1) {
    "use strict";
    var LayerContentExample = (function (_super) {
        __extends(LayerContentExample, _super);
        function LayerContentExample() {
            _super.call(this);
            this.state = {
                time: new Date().toLocaleTimeString()
            };
        }
        LayerContentExample.prototype.componentDidMount = function () {
            var _this = this;
            this._async.setInterval(function () { return _this.setState({ time: new Date().toLocaleTimeString() }); }, 1000);
        };
        LayerContentExample.prototype.render = function () {
            return (React.createElement("div", {className: 'LayerExample-content ms-u-scaleUpIn100'}, 
                React.createElement("div", {className: 'LayerExample-textContent'}, this.context.message), 
                React.createElement("div", null, this.state.time)));
        };
        LayerContentExample.contextTypes = {
            message: React.PropTypes.string
        };
        return LayerContentExample;
    }(index_1.BaseComponent));
    exports.LayerContentExample = LayerContentExample;
    var LayerBasicExample = (function (_super) {
        __extends(LayerBasicExample, _super);
        function LayerBasicExample() {
            _super.call(this);
            this.state = {
                showLayer: false
            };
        }
        LayerBasicExample.prototype.getChildContext = function () {
            return {
                'message': 'Hello world.'
            };
        };
        LayerBasicExample.prototype.render = function () {
            var _this = this;
            var showLayer = this.state.showLayer;
            return (React.createElement("div", null, 
                React.createElement(index_1.Checkbox, {label: 'Wrap the content box belowed in a Layer', checked: showLayer, onChange: function (ev, checked) { return _this.setState({ showLayer: checked }); }}), 
                showLayer ? (React.createElement(index_1.Layer, null, 
                    React.createElement(LayerContentExample, null)
                )) : (React.createElement(LayerContentExample, null))));
        };
        LayerBasicExample.childContextTypes = {
            message: React.PropTypes.string
        };
        return LayerBasicExample;
    }(index_1.BaseComponent));
    exports.LayerBasicExample = LayerBasicExample;
});
